### PR TITLE
feat: add i18n support for 5 languages (en, es, fr, de, zh)

### LINKED ISSUE #93: Multi-Language CLI Support
## Description
Support multiple languages for CLI interface (Spanish, French, German, Chinese).

## Requirements
- Internationalization framework
- Language selection
- Message catalog management
- Date/time localization
- Number formatting
- Translation management

## Acceptance Criteria
- [ ] i18n framework implemented
- [ ] At least 3 languages supported
- [ ] Language auto-detection
- [ ] User language preference
- [ ] Complete message translation
- [ ] Locale-aware formatting
- [ ] Unit tests included (>80% coverage)
- [ ] Documentation for translators

## Example Usage
```bash
$ cortex config language es
✓  Idioma cambiado a español

$ cortex install docker
🔍 Analizando sistema...
✓  Docker instalado correctamente

$ cortex config language zh
✓  语言已更改为中文
```

**Skills:** Python, i18n, localization, gettext
**Bounty:** $25 upon merge
**Priority:** Low

### PR DESCRIPTION
**Pull request by RIVALHIDE**
## Related Issue
Closes #93 

## Summary
Adds comprehensive multi-language support (i18n) to the Cortex CLI with proper localization, formatting, tests, and documentation.

### Supported Languages
| Code | Language | Native Name |
|------|----------|-------------|
| `en` | English | English |
| `es` | Spanish | Español |
| `fr` | French | Français |
| `de` | German | Deutsch |
| `zh` | Chinese | 中文 |

### Key Features
- ✅ Scalable i18n framework with externalized messages in YAML catalogs
- ✅ Language selection via `cortex config language <code>` or `cortex --set-language "English"`
- ✅ OS language auto-detection from `LANG`, `LC_ALL`, `LANGUAGE` environment variables
- ✅ Persistent language preferences stored in `~/.cortex/preferences.yaml`
- ✅ Locale-aware formatting for dates, times, and numbers using Babel
- ✅ Graceful fallback to English for missing translations
- ✅ Debug mode (`CORTEX_I18N_DEBUG=1`) to show translation keys
- ✅ 100% CLI messages translated (no English leaks)
- ✅ Unit tests with >80% coverage

### Set language by code
cortex config language es

### Set language by name (human-readable)
cortex --set-language "Spanish"

### View current language info
cortex config language --info

### List available languages
cortex config language --list

### Use auto-detection
cortex config language --auto

### Clear preference (use OS default)
cortex config language --clear

## Files Changed

| File | Change | Reason |
|------|--------|--------|
| `pyproject.toml` | Modified | Added `cortex.i18n` package, locale YAML files as package data, upgraded setuptools to >=64.0 for editable install support |
| `MANIFEST.in` | Modified | Added `recursive-include cortex/i18n/locales *.yaml` to include translations in PyPI distribution |
| `cortex/cli.py` | Modified | Integrated i18n throughout CLI - replaced hardcoded strings with `t()` calls, added `config language` subcommand, added `--set-language` global flag |
| `cortex/i18n/__init__.py` | Added | Module initialization, exports core i18n API (`t`, `set_language`, `get_language`, etc.) |
| `cortex/i18n/translator.py` | Added | Core translation engine - loads YAML catalogs, handles interpolation, manages fallback to English |
| `cortex/i18n/config.py` | Added | Language preference persistence to `~/.cortex/preferences.yaml`, preference resolution order |
| `cortex/i18n/detector.py` | Added | OS language auto-detection from environment variables (`LANG`, `LC_ALL`, `LANGUAGE`, `LC_MESSAGES`) |
| `cortex/i18n/formatter.py` | Added | Locale-aware formatting for numbers, dates, time-ago, durations, and file sizes using Babel |
| `cortex/i18n/locales/en.yaml` | Added | English message catalog (source language) - 376 lines |
| `cortex/i18n/locales/es.yaml` | Added | Spanish translations - 100% coverage |
| `cortex/i18n/locales/fr.yaml` | Added | French translations - 100% coverage |
| `cortex/i18n/locales/de.yaml` | Added | German translations - 100% coverage |
| `cortex/i18n/locales/zh.yaml` | Added | Chinese (Simplified) translations - 100% coverage |
| `tests/test_i18n.py` | Added | Comprehensive test suite - translator, config, detector, formatter, CLI integration (>80% coverage) |
| `docs/i18n-guide.md` | Added | Documentation for translators and developers on using/extending the i18n system |

## AI Disclosure
- [x] AI/IDE/Agents used (please describe below)

Used **Cursor AI (Claude opus 4.5)** to assist with:
- Writing comprehensive unit tests
- Integrating i18n into existing CLI commands
- Debugging and fixing linting/type-checking issues
- Creating translator documentation

## Checklist
- [x] PR title follows format: `type(scope): description` or `[scope] description`
- [x] Tests pass (`pytest tests/`)
- [ ] MVP label added if closing MVP issue
- [x] Update "Cortex -h" (if needed)

### Language Resolution Order
1. `CORTEX_LANGUAGE` environment variable (highest priority)
2. User preference in `~/.cortex/preferences.yaml`
3. OS-detected language from `LANG`/`LC_ALL`
4. Default: English

### Demo video 
[Screencast from 14-01-26 07:13:17 PM IST.webm](https://github.com/user-attachments/assets/01579727-1dbc-4081-9496-780920e8175c)

## Summary by CodeRabbit

* **New Features**
  * Full i18n support (en, es, fr, de, zh): localized CLI messages, translation catalogs, translator API, and locale-aware formatting/parsing.
  * CLI language management: list/info/set, --language/--set-language early switch, auto-detect and persist preference, and reset.

* **Documentation**
  * Added comprehensive i18n guide covering usage, adding languages, CLI commands, and troubleshooting.

* **Tests**
  * Extensive unit and integration tests for translations, detection, persistence, and formatting; e2e assertions relaxed for localized outputs.

* **Chores**
  * Packaging updated to include locale files.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

**Review by Anshgrover23 — CHANGES_REQUESTED** [review]
@RIVALHIDE Kindly address bot comments and then create a new Issue regarding auto-translation thing.

**Review by mikejmorgan-ai — APPROVED** [review]
## Review: APPROVED ✅

**i18n Support - 5 Languages (PR #593)**

### Code Quality
- ✅ Thread-safe file operations with proper locking (threading.Lock + fcntl.flock)
- ✅ TOCTOU race condition properly addressed - exists() check and file read within critical section
- ✅ Formatter singleton correctly syncs language on each access
- ✅ Comprehensive error handling for malformed YAML, empty files, invalid types
- ✅ Well-documented code with clear docstrings

### Test Coverage
- ✅ 89 tests all passing
- ✅ Tests cover: translator, config, detector, formatter, CLI integration
- ✅ Edge cases tested: malformed YAML, null values, invalid types, whitespace files
- ✅ All 5 languages verified: English, Spanish, French, German, Chinese

### Architecture
- ✅ Clean separation: translator.py, formatter.py, config.py, detector.py
- ✅ Single source of truth for supported languages (SUPPORTED_LANGUAGES dict)
- ✅ Proper fallback chain: env var → config file → OS detection → default

### Notes for CI
- Requires Python 3.10+ due to union type syntax (`type | None`)
- Ensure CI runs with compatible Python version

Ready for merge once Ansh approves.

Reviewed-by: Claude Code

File: cortex/i18n/locales/fr.yaml [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+# Indicateurs de progression
+# =============================================================================
+progress:
+  analyzing_requirements: "Analyse des exigences système..."
+  updating_packages: "Mise à jour de la liste des paquets..."
+  downloading: "Téléchargement..."
+  extracting: "Extraction..."
+  configuring: "Configuration..."
+  cleaning_up: "Nettoyage..."
+  completed_in: "Terminé en {seconds} secondes"
```
[the remark this answers is not in the document]
Comment (Anshgrover23) [reply cortex/i18n/locales/fr.yaml]: @RIVALHIDE Kindly address this one.
